shell/components/form/GithubPicker.vue

```diff
@@ -1,343 +1,355 @@
-    filePattern: {
-      type:    String,
-      default: null,
-    },
-
-    preferredFile: {
-      type:    String,
-      default: null,
-    },
-
-    fileKey: {
-      type:    String,
-      default: 'dockerfile',
-    }
+  components: {
+    LabeledSelect,
+    SortableTable,
+    RadioButton,
+    LabeledInput,
   },
 
   data() {
-    return {
-      repos:       [],
-      branches:    [],
-      files:       [],
+    const eventHeaders = [
+      {
+        name:  'index',
+        label: 'Chose',
+      },
+      {
+        name:  'sha',
+        label: 'SHA',
+        value: 'sha',
+      },
+      {
+        name:  'author',
+        label: 'Author',
+        value: 'author.login',
+      },
+      {
+        name:  'message',
+        label: 'Message',
+        value: 'message',
+      },
+    ];
 
+    return {
       loadingRecentRepos: true,
       loadingBranches:    true,
       loadingFiles:       true,
-
-      selectedRepo:   null,
-      selectedBranch: null,
-      selectedFile:   null
+      loadingCommits:     true,
+
+      hasError: {
+        username: false,
+        repo:     false,
+        branch:   false,
+        commits:  false,
+      },
+      showSelections: false,
+      oldUsername:    null,
+
+      repos:    [],
+      branches: [],
+      commits:  [],
+
+      selectedUsername: null,
+      selectedRepo:     null,
+      selectedBranch:   null,
+      selectedCommit:   false,
+      eventHeaders,
     };
   },
 
-  computed:   {
-    ...mapState({
-      scopes:      state => state.github.scopes,
-      recentRepos: state => state.github.repos,
-    }),
-
-    hasPrivate() {
-      return this.scopes.includes('repo');
+  computed: {},
+  methods:  {
+    reset() {
+      this.selectedRepo = null;
+      this.selectedBranch = null;
+      this.selectedCommit = false;
     },
+    async fetchRepos() {
+      try {
+        if (this.selectedUsername.length) {
+          this.selectedRepo = null;
 
-    repoPlaceholder() {
-      if ( this.loadingRecentRepos ) {
-        return 'Loading...';
-      } else {
-        return 'Select a Repository...';
-      }
-    },
+          const res = await this.$store.dispatch('github/fetchRecentRepos', { username: this.selectedUsername });
 
-    branchPlaceholder() {
-      if ( this.selectedRepo ) {
-        if ( this.loadingBranches ) {
-          return 'Loading...';
-        } else {
-          return 'Select a Branch...';
+          if (res?.message) {
+            this.hasError.repo = true;
+          } else {
+            this.repos = res;
+            this.hasError.repo = false;
+          }
+
+          // Reset selections once username changes
+          if (this.oldUsername !== this.selectedUsername) {
+            this.oldUsername = this.selectedUsername;
+
+            return this.reset();
+          }
         }
-      } else {
-        return 'Select a Repository First';
+      } finally {
```

Comment:
Ok, slightly missunderstood.

The github store `fetchGithubAPI` has the catch block

```
  } catch (error) {
    if (error.response) {
      return error.response;
    }
  }
```
This will only return a string in some cases (if error.response is missing `undefined` will return). In github picker we only check for `res?.message`, which might have overlap with something returned by github. 

If would probably better to
- change `fetchGithubAPI` to find a suitable error message (error.response.message or undefined) and throw that
- in apiList catch this, if undefined throw a custom message ('failed to fetch x')
- in github picker catch the guaranteed message and set error etc